lib/i18n/backend/http/etag_http_client.rb

```diff
@@ -6,24 +6,49 @@ class Http
       class I18n::Backend::Http::EtagHttpClient
         def initialize(options)
           @options = options
+          @statsd_client = options[:statsd_client]
         end
 
         def download(path, etag:)
           @client ||= Faraday.new(@options[:host])
+          start     = Time.now
+
           response = @client.get(path) do |request|
             request.headers.merge!(@options[:headers]) if @options[:headers]
             request.headers["If-None-Match"] = etag if etag
-            request.options[:timeout] = @options[:http_read_timeout]
-            request.options[:open_timeout] = @options[:http_open_timeout]
+            request.options[:timeout]        = @options[:http_read_timeout]
+            request.options[:open_timeout]   = @options[:http_open_timeout]
           end
 
+          record :timing, time: (Time.now - start).to_i, tags: {path: path}
+
           case response.status
-          when 200 then [response.body, response['ETag']]
-          when 304 then nil
+          when 200
+            record :success, tags: {status_code: response.status, path: path}
```

Comment:
I could make it nicer, but didn't want to get too zendesk specific. `path` is implemented by the consumer so hard to say, but for use it is limited.